### ISSUE TITLE
EmptyLineSeparator does not detect multiple blank lines

### ISSUE THREAD

**Issue by serpro69** [checkstyle/checkstyle#14763]
- [x] I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
- [x] I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
- [x] I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

The following java file clearly has too many blank lines, which should have been reported by checkstyle's EmptyLineSeparator

```java
package no.wlcm.tech.service.impl;

import no.wlcm.tech.model.Greeting;
import no.wlcm.tech.service.GreetingService;
import org.springframework.stereotype.Service;

@Service
public class GreetingServiceImpl implements GreetingService {








    @Override
    public Greeting greetingFor(String name) {



        return (name != null && !name.isEmpty()) ? new Greeting("Welcome to Welcome, " + name + "!") :
           new Greeting("Welcome to Welcome!");
    }



}
```

Running latest version of checkstyle does not show a warning/error for this issue.

```bash
➜ java -jar /tmp/checkstyle-10.15.0-all.jar -c /google_checks.xml src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java
Starting audit...
[WARN] /home/sergio/Projects/wlcmtech/template-spring-boot/src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java:7:1: Missing a Javadoc comment. [MissingJavadocType]
[WARN] /home/sergio/Projects/wlcmtech/template-spring-boot/src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java:17:5: 'method def modifier' has incorrect indentation level 4, expected level should be 2. [Indentation]
[WARN] /home/sergio/Projects/wlcmtech/template-spring-boot/src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java:22: Line is longer than 100 characters (found 102). [LineLength]
[WARN] /home/sergio/Projects/wlcmtech/template-spring-boot/src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java:22:9: 'method def' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /home/sergio/Projects/wlcmtech/template-spring-boot/src/main/java/no/wlcm/tech/service/impl/GreetingServiceImpl.java:24:5: 'method def rcurly' has incorrect indentation level 4, expected level should be 2. [Indentation]
Audit done.

```

Please ignore the indentation warning, the issue is not about indentation.

I've checked the google config and it has EmptyLineSeparator - https://github.com/checkstyle/checkstyle/blob/5e2a2f096a7564984b846619268435c639096458/src/main/resources/google_checks.xml#L142 , so it's clearly not the config file issue.

**Comment by mahfouz72** [comment]
I think this is an expected behavior please read [EmptyLineSepartorCheck](https://checkstyle.org/checks/whitespace/emptylineseparator.html#EmptyLineSeparator) to understand more about how the check works

**Comment by serpro69** [comment]
Hi @mahfouz72 ,
Thanks for the comment.
So you mean there's no way to check and report multi-blank-line violations like in the above example?

**Comment by mahfouz72** [comment]
> you mean there's no way to check and report multi-blank-line violations like in the above example?

No, you can do this checkstyle is highly configurable. but then you need to create a custom config file and change the default value of the properties of the check to achieve what you want

**Comment by serpro69** [comment]
Oh I see. Thanks a lot for the info and for providing an example!

**Comment by romani** [comment]
Users should not read code, if we have gap in doc, lets extend it

**Comment by MANISH-K-07** [comment]
@romani , we already have 2 dedicated examples that show usage of these properties in config mentioned at https://github.com/checkstyle/checkstyle/issues/14763#issuecomment-2039859989

https://checkstyle.org/checks/whitespace/emptylineseparator.html#Example4-config
https://checkstyle.org/checks/whitespace/emptylineseparator.html#Example5-config
